packages/edit-site/src/components/block-editor/use-site-editor-settings.js

```diff
@@ -80,16 +78,12 @@ export default function useSiteEditorSettings( templateType ) {
 			inserterMediaCategories,
 			__experimentalBlockPatterns: blockPatterns,
 			__experimentalBlockPatternCategories: blockPatternCategories,
-			// Template locking must be explicitly "unset" for non-navigation entities.
-			templateLock: isNavigationFocusMode ? 'insert' : false,
-			template: isNavigationFocusMode
-				? [ [ 'core/navigation', {}, [] ] ]
-				: false,
+			// Some Site Editor entities (e.g. `wp_navigation`) may utilise
+			// template locking in their settings. Therefore this must be
+			// explicitly "unset" to avoid the template locking UI being
+			// active for all entities.
+			templateLock: false,
+			template: false,
```

Comment:
Isn't that the default values though?

Reply:
Let me test. There was a bug when I _didn't_ unset this previously.

Reply:
https://github.com/WordPress/gutenberg/assets/444434/c25a8cb6-1856-4af0-be49-927c3ccb07bf

Here is the bug. Switching between Navigation and other entities causes the template lock to remain in place.

I guess there might be something up with the memoization. However, previously I check the `settings` being passed to the editor and `templateLock` isn't included _at all_ in the default settings and the UI is still locked 😕

Reply:
Definitely a bug somewhere in the `BlockEditorProvider`. I'm guessing when the setting change, we shouldn't "merge" with previous settings but instead replace the whole object + defaults. I think we should solve that separately from this PR. It's fine to leave a not in the comment about that bug.

Reply:
Bug report in https://github.com/WordPress/gutenberg/issues/51489